apps/nextra/pages/en/build/smart-contracts/linter.mdx

```diff
@@ -12,6 +12,15 @@ This tool is currently in beta, so please try it out and submit [bugs and feedba
 
 
 ## Lint Checks
+### `almost_swapped`
+Checks for expression patterns that look like a failed swap attempt and notifies the user. These patterns are likely erroneous code. This currently only detects simple access patterns including assignments to a variable or a field of a struct. Examples include:
+- `a = b; b = a;` which can be correctly swapped with `(a, b) = (b, a);`
+- `a.x = b.x; b.x = a.x;` which can be correctly swapped with `(a.x, b.x) = (b.x, a.x);`
+
+Note that this does not detect patterns involving non-builtin functions, including vector operations, nor does this detect patterns that deal with global storage. For example, the following are out of scope of this lint check:
```

Comment:
I think we should leave out examples of what is not caught, we already say this currently only detects simple access patterns.

Reply:
Okay. I can delete. I added this because I figure that "simple access patterns" is not really a well defined term.

Reply:
The terminology "non-builtin functions, including vector operations" may not be obvious to the user (builtin functions may be specific to compiler internals).

Reply:
> I added this because I figure that "simple access patterns" is not really a well defined term.

I figured that was your intention. But reading it from a user's perspective, I think it is too much programmatic detail. But if you feel strongly, feel free to leave it in, but address the other comments.